### PR TITLE
deprecate function `get_default_device` in `tools/base.py`

### PR DESCRIPTION
**Pull request by ji-huazhong**
# What does this PR do?

As per title.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by muellerzr — APPROVED** [review]
Thanks! @ArthurZucker should this be on a deprecation cycle for the function name? Otherwise looks great to me

**Review by ArthurZucker — COMMENTED** [review]
Yes! As @muellerzr mentions, I think we should leave it for now, with a deprecation warning mention that we'll remove it soon and that the accelerate version should be used!

File: src/transformers/tools/base.py [hunk 1]
```diff
@@ -599,6 +599,10 @@ def fn(*args, **kwargs):
 
 # TODO: Migrate to Accelerate for this once `PartialState.default_device` makes its way into a release.
 def get_default_device():
+    logger.warning(
+        "`get_default_device` is deprecated and will be replaced with Accelerate's `PartialState.defualt_device` "
```
Comment (muellerzr) [inline src/transformers/tools/base.py:603]:
```suggestion
        "`get_default_device` is deprecated and will be replaced with `accelerate`'s `PartialState().device` "
```

Let's use the right API please :)

**Review by ArthurZucker — APPROVED** [review]
LGTM with the proposed changes!

**Comment by statelesshz** [comment]
this PR is ready to be merged :)

**Comment by statelesshz** [comment]
ping @muellerzr again :)

**Comment by statelesshz** [comment]
cc @amyeroberts

**Review by amyeroberts — APPROVED** [review]
Thanks for adding!

**Pull request merged**